pkg/flowkit/services/events.go

```diff
@@ -50,46 +50,107 @@ func NewEvents(
 	}
 }
 
-// Get queries for an event by name and block range.
-func (e *Events) Get(name string, start string, end string) ([]client.BlockEvents, error) {
-	if name == "" {
-		return nil, fmt.Errorf("cannot use empty string as event name")
+func (e *Events) CalculateStartEnd(start uint64, end uint64, last uint64) (uint64, uint64, error) {
+
+	if start == 0 {
+		latestBlock, err := e.gateway.GetLatestBlock()
+		if err != nil {
+			return 0, 0, err
+		}
+		return latestBlock.Height - last, latestBlock.Height, nil
 	}
 
-	startHeight, err := strconv.ParseUint(start, 10, 64)
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse start height of block range: %v", start)
+	if end == 0 {
+		latestBlock, err := e.gateway.GetLatestBlock()
+		if err != nil {
+			return 0, 0, err
+		}
+		return start, latestBlock.Height, nil
 	}
 
+	if end < start {
+		return 0, 0, fmt.Errorf("cannot have end height (%d) of block range less that start height (%d)", end, start)
+	}
+	return start, end, nil
+}
+
+func makeEventQueries(events []string, startHeight uint64, endHeight uint64, blockCount uint64) []client.EventRangeQuery {
+	var queries []client.EventRangeQuery
+	for startHeight <= endHeight {
+		suggestedEndHeight := startHeight + blockCount - 1 //since we are inclusive
+		endHeight := endHeight
```

Comment:
Not sure why is this necessary?

Reply:
flow events get A.1654653399040a61.FlowToken.TokensDeposited 16128771 16129021
The difference here is 250, but the blockchain says 
❌ Command Error: client: rpc error: code = Unknown desc = requested block range (251) exceeded maximum (250)

Since fetching events are inclusive to the start block.

Reply:
No, I mean this statement specifically `endHeight := endHeight`

Reply:
Remove it. legacy from when i has the modifier on line 81 on that line instead.

Reply:
I am not completely sure why but this line needs to be here, if it is not it will not work, It will use the endHeight from the previous iteration of the loop. Try to remove it and observe the tests.

Reply:
There is a difference if that statement gets deleted because you are actually creating another variable within for block that is different from the value the for loop uses. That is why I think it would be better to either change the naming or rethink how this function is written.